osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/MigratePlaylistScoresToSoloScoresCommand.cs

```diff
@@ -0,0 +1,199 @@
+                                });
+
+                                soloScoreInfo.StartedAt = score.started_at;
+                                soloScoreInfo.EndedAt = score.started_at;
+                                soloScoreInfo.BeatmapID = (int)score.beatmapId;
+
+                                insertId = await db.InsertAsync(new SoloScore
+                                {
+                                    user_id = (int)score.userId,
+                                    beatmap_id = (int)score.beatmapId,
+                                    ruleset_id = item.ruleset_id,
+                                    preserve = true,
+                                    ScoreInfo = soloScoreInfo,
+                                    created_at = (DateTimeOffset)score.created_at!,
+                                    updated_at = (DateTimeOffset)score.updated_at!,
+                                });
+                            }
+
+                            await db.ExecuteAsync("INSERT INTO multiplayer_score_links (user_id, room_id, beatmap_id, playlist_item_id, score_id, created_at, updated_at) VALUES (@userId, @roomId, @beatmapId, @playlistItemId, @scoreId, @createdAt, @updatedAt)", new
+                            {
+                                score.userId,
+                                score.roomId,
+                                score.beatmapId,
+                                score.playlistItemId,
+                                scoreId = insertId,
+                                createdAt = score.ended_at,
+                                updatedAt = score.ended_at
+                            });
+
+                            Console.WriteLine();
+                        }
+                    }
+                }
+            }
+
+            Console.WriteLine("Finished.");
+            return 0;
+        }
+
+        private static HitResult getMaxJudgementFor(HitResult hitResult, HitResult max)
+        {
+            switch (hitResult)
+            {
+                case HitResult.Miss:
+                case HitResult.Meh:
+                case HitResult.Ok:
+                case HitResult.Good:
+                case HitResult.Great:
+                case HitResult.Perfect:
+                    return max;
+
+                case HitResult.SmallTickMiss:
+                case HitResult.SmallTickHit:
+                    return HitResult.SmallTickHit;
+
+                case HitResult.LargeTickMiss:
+                case HitResult.LargeTickHit:
+                    return HitResult.LargeTickHit;
+            }
+
+            return HitResult.IgnoreHit;
+        }
+
+        // ReSharper disable InconsistentNaming
+        public class MultiplayerScore
+        {
+            public ulong id { get; set; }
+            public uint userId { get; set; }
+            public ulong roomId { get; set; }
+            public ulong playlistItemId { get; set; }
+            public uint beatmapId { get; set; }
```

Comment:
> Additionally, after applying the above, the following:

This might have changed since when you reviewed, but it doesn't look like changes should be applied here.